### ISSUE TITLE
SHSH2 Blob Specification

### ISSUE THREAD

**Issue by iPodHacks142** [tihmstar/futurerestore#97]
When running futurerestore, a better way to specify blobs would be to specify a folder full of them, rather than the individual blob files.

**Comment by boobang** [comment]
you can do with typing more "-t" command...
Example: -t 1.shsh2 -t 2.shsh2 -t 3.shsh2 -t 4.shsh2  .... This should be helpful..

**Comment by iPodHacks142** [comment]
I'm very aware of this. What I am saying is rather that specifying each blob individually, you would specify a folder full of them. This way you wouldn't have to type all of them.

**Comment by RoyalGraphX** [comment]
I would like this feature to be added as i save 1,000 shsh2 blobs for every version since 9.3.3 in seperate folders and this would be painful to do -t 1,000 times.

**Comment by leuldereje** [comment]
Shouldn't this be a pull request?

**Comment by tihmstar** [comment]
I see that this kinda makes sense, but i probably won't implement this anytime soon, because it's simply a lot of work :/
Right now when a user specifys a shsh file, i assume it is *compatible* and just work with it. If it turns out later to be incompatible (eg ECID doesn't match), i just blame the user for that and exit futurerestore. 
By specifying a dir i'd have to scan each file in that dir (and any subdirs recursively i guess?) to actually match the current restore and only add them if they are good. 
That means checking ECID, checking restore mode (update/erase) and much more. 
Then i'd have to cover lots of edge cases which all in all is simply a lot of work…

Instead what you can do is scan that dir with a few bash commands
'find /path/to/shshfiles -name "DEVICE*.shsh*" > a.txt'
(Or some similar command which gives you all blobs you need)
then open that file and search&replace "DEVICE" with "-t DEVICE" and finally
'tr '\n' ' ' < a.txt > b.txt'
Then when you run futurerestore do something like:
'futurerestore $(cat b.txt) --latest-baseband --latest-sep IPSWFILE.ipsw'
So basically that's what i'm doing. 

Don't quote me on every command, they might be slightly wrong, but that's just so you get the basic idea.

**Comment by RoyalGraphX** [comment]
I can see how that would be annoying to implement to be completely honest with you too.
